src/net_doh.c

```diff
@@ -872,16 +871,20 @@ static ssize_t perf__doh_recv(struct perf_net_socket* sock, void* buf, size_t le
 
     if (self->http2 &&
         self->http2->dnsmsg_completed) {
-        if (self->http2->dnsmsg_at > len) {
+        if (self->http2->dnsmsg_at + 2 > len) {
             perf_log_warning("failed to process result - DNS response size");
             PERF_UNLOCK(&self->lock);
             return -1;
         }
 
-        // debugx("copying to recv buf - len: %d", self->http2->dnsmsg_at);
-        memcpy(buf, self->http2->dnsmsg, self->http2->dnsmsg_at);
+        // debugx("copying to recv buf - len: %ld", self->http2->dnsmsg_at);
+        uint16_t qid = htons(self->qid);
+        memcpy(buf, &qid, 2);
```

Comment:
No no no! What are you doing? Why are you injecting this?!?

Reply:
@jelu - sorry, this is a spill from feature/doh-reconnect where I was tracking down why net_recv() packet_header was having zeros all the time on receive and had to debug it. The reconnect stuff got merged, this was supposed to be for testing.

Latest commit clears this back, as it clears most of the debug code left overs.

Reply:
also, I left most of the nghttp2 callbacks in place even if they are not required for testing purposes on the http2 workflow when testing against different endpoints.